go/vt/sqlparser/comments.go

```diff
@@ -322,86 +322,44 @@ func (d *CommentDirectives) GetString(key string, defaultVal string) (string, bo
 
 // MultiShardAutocommitDirective returns true if multishard autocommit directive is set to true in query.
 func MultiShardAutocommitDirective(stmt Statement) bool {
-	var comments *ParsedComments
-	switch stmt := stmt.(type) {
-	case *Insert:
-		comments = stmt.Comments
-	case *Update:
-		comments = stmt.Comments
-	case *Delete:
-		comments = stmt.Comments
-	}
-	return comments != nil && comments.Directives().IsSet(DirectiveMultiShardAutocommit)
+	return checkDirective(stmt, DirectiveMultiShardAutocommit)
 }
 
 // SkipQueryPlanCacheDirective returns true if skip query plan cache directive is set to true in query.
 func SkipQueryPlanCacheDirective(stmt Statement) bool {
-	var comments *ParsedComments
-	switch stmt := stmt.(type) {
-	case *Select:
-		comments = stmt.Comments
-	case *Insert:
-		comments = stmt.Comments
-	case *Update:
-		comments = stmt.Comments
-	case *Delete:
-		comments = stmt.Comments
-	}
-	return comments != nil && comments.Directives().IsSet(DirectiveSkipQueryPlanCache)
+	return checkDirective(stmt, DirectiveSkipQueryPlanCache)
```

Comment:
@harshit-gangal this is changing the behaviour for DMLs in the query cache. Are we intentionally skipping the cache?

Reply:
why do you think DMLs are handled differently here?

Reply:
May be I am missing something.

Reply:
talked it over. I was confused and wrote DML when I meant DDL